### PR TITLE
File Dialog: Add dynamic preview

### PR DESCRIPTION
**Pull request by hernanmd**
This PR adds the following features to File Dialog support:

- Add a method to find previewers providing its extension or a FileReference.
- Add GIF previewer
- Add previewer tests.

### CODE REVIEW

File: src/NewTools-FileBrowser/StFileBrowserTextBasedContentPreviewer.class.st [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	^ true
+]
+
+{ #category : 'private' }
+StFileBrowserTextBasedContentPreviewer >> numberOfBytesToCheckForBinaryOrText [
+	
+	^ 100
+]
+
+{ #category : 'as yet unclassified' }
```
Comment (jecisc) [inline src/NewTools-FileBrowser/StFileBrowserTextBasedContentPreviewer.class.st]:
```suggestion
{ #category : 'accessing' }
```

**Review by jecisc — CHANGES_REQUESTED** [review]

File: src/NewTools-FileBrowser/StFileBrowserTextBasedContentPreviewer.class.st (hunk 1 above)
Comment (Ducasse) [reply src/NewTools-FileBrowser/StFileBrowserTextBasedContentPreviewer.class.st]: @hernanmd can you rewrite it so that we push it then?

Comment (hernanmd) [reply src/NewTools-FileBrowser/StFileBrowserTextBasedContentPreviewer.class.st]: Ok, done

**Comment by Ducasse** [comment]
@hernanmd you have 

```
✗ #testObsoleteClasses (837ms)
TestFailure: Obsolete classes remaining: an Array(AnObsoleteStFileBrowserContentPreviewer)
ReleaseTest(TestAsserter)>>assert:description:resumable:
ReleaseTest(TestAsserter)>>assert:description:
ReleaseTest>>testObsoleteClasses ...assert: obsoleteClasses isEmpty
		description: [
			String streamContents: [ :s|
				s
					nextPutAll: 'Obsolete classes remaining: ';
					print: obsoleteClasses ]]
ReleaseTest(TestCase)>>performTest
```

**Comment by hernanmd** [comment]
Hi @jecisc, I tried to reproduce locally the failing Pharo-Integration tests but even after fixing a Sindarin loading error, the tests (`testNoUnusedClassVariablesLeft`and `testObsoleteClasses`) are green. This is how I load in a new Pharo 12 image:

```smalltalk
Metacello new
	baseline: 'NewTools';
	repository: 'github://hernanmd/NewTools:add_dynamic_preview/src';
	load.
```

I also tried running through smalltalkCI but it was worst as Spec changes were missing:

```bash
bin/smalltalkci --headful -s Pharo64-12 ../git-repositories/NewTools/.smalltalk.ston
```

Do you know what else to try?

**Comment by hernanmd** [comment]
> Hi @jecisc, I tried to reproduce locally the failing Pharo-Integration tests but even after fixing a Sindarin loading error, the tests (`testNoUnusedClassVariablesLeft`and `testObsoleteClasses`) are green. This is how I load in a new Pharo 12 image:
> 
> ```smalltalk
> Metacello new
> 	baseline: 'NewTools';
> 	repository: 'github://hernanmd/NewTools:add_dynamic_preview/src';
> 	load.
> ```
> 
> I also tried running through smalltalkCI but it was worst as Spec changes were missing:
> 
> ```shell
> bin/smalltalkci --headful -s Pharo64-12 ../git-repositories/NewTools/.smalltalk.ston
> ```
> 
> Do you know what else to try?

Ok, I managed to load the PR in the TravisCI.image and discovered lot of methods were missing, causing unreferenced variables and obsolete classes. I tried to rename the classes but still the Obsolete are introduced in the Pharo Integration CI job. 

So this could mean code importing bug?

**Comment by hernanmd** [comment]
Finally. Errors were related to the `{} + TraitedClass` leftovers.
Now reported errors are not related:

```
LinkInstallerTest
 ✗ #testSlotOrVarLinksRemainAfterMethodModificationForObject (21ms)
 ✗ #testSlotOrVarLinksRemainAfterMethodModification (16ms)

RBRemoveProtocolTransformationTest
 ✗ #testTransform (1ms)

SocketStreamTest
 ✗ #testFlushLargeMessageOtherEndClosed (2ms)

RSDependencyTest
 ✗ #testDependencies (3ms)

StDebuggerTest
 ✗ #testUpdateExtensionSubscription (14ms)

OCTargetCompilerTest
 ✗ #testModifiedReturnFromClass (0ms)
  Executed 38664 Tests with 5 Failures and 2 Errors in 503.26s.
To reproduce the failed build locally, download smalltalkCI, and try to run something like:
	bin/smalltalkci -s Pharo64-12 --headfull /path/to/.smalltalk.ston
```

**Comment by Ducasse** [comment]
Tx

**Pull request merged**